core/include/drivers/gpio.h

```diff
@@ -166,16 +199,39 @@ TEE_Result gpio_dt_alloc_pin(struct dt_pargs *pargs, struct gpio **gpio);
 TEE_Result gpio_dt_get_by_index(const void *fdt, int nodeoffset,
 				unsigned int index, const char *gpio_name,
 				struct gpio **gpio);
+
+/**
+ * gpio_dt_cfg_by_index() - Get a GPIO controller at a specific index in
+ * 'gpios' property and configure with provided flags
+ *
+ * @fdt: Device tree to work on
+ * @nodeoffset: Node offset of the subnode containing a 'gpios' property
+ * @index: GPIO pin index in '*-gpios' property
+ * @gpio_name: Prefix of the GPIO properties in device tree, can be NULL
+ * @flags: requester flags of GPIO
+ * @gpio: Output GPIO pin reference upon success
+ *
```

Comment:
Add a note about the property allowed to be "gpio", "gpios", "<prefix>-gpio" or "<prefix>-gpios".